api/v1/mongodbcommunity_types.go

```diff
@@ -736,29 +761,92 @@ func (m *MongoDBCommunity) AutomationConfigArbitersThisReconciliation() int {
 	})
 }
 
+// GetOptionsString return a string format of the connection string
+// options that can be appended directly to the connection string.
+//
+// Only takes into account options for the resource and not any user.
+func (m *MongoDBCommunity) GetOptionsString() string {
+	generalOptionsMap := m.Spec.AdditionalConnectionStringConfig.Object
+	optionValues := make([]string, len(generalOptionsMap))
+	i := 0
+
+	for key, value := range generalOptionsMap {
+		optionValues[i] = fmt.Sprintf("%s=%v", key, value)
+		i += 1
+	}
+
+	optionValues = optionValues[:i]
+
+	optionsString := ""
+	if i > 0 {
+		optionsString = "&" + strings.Join(optionValues, "&")
+	}
+	return optionsString
+}
+
+// GetUserOptionsString return a string format of the connection string
+// options that can be appended directly to the connection string.
+//
+// Takes into account both user options and resource options.
+// User options will override any existing options in the resource.
+func (m *MongoDBCommunity) GetUserOptionsString(user scram.User) string {
+	generalOptionsMap := m.Spec.AdditionalConnectionStringConfig.Object
+	userOptionsMap := user.ConnectionStringOptions
+	optionValues := make([]string, len(generalOptionsMap)+len(userOptionsMap))
+	i := 0
+	for key, value := range userOptionsMap {
+		optionValues[i] = fmt.Sprintf("%s=%v", key, value)
+		i += 1
+	}
+
+	for key, value := range generalOptionsMap {
+		_, ok := userOptionsMap[key]
+		if !ok {
+			optionValues[i] = fmt.Sprintf("%s=%v", key, value)
+			i += 1
+		}
+	}
+
+	optionValues = optionValues[:i]
+
+	optionsString := ""
+	if i > 0 {
+		optionsString = "&" + strings.Join(optionValues, "&")
+	}
+	return optionsString
+}
+
 // MongoURI returns a mongo uri which can be used to connect to this deployment
 func (m *MongoDBCommunity) MongoURI(clusterDomain string) string {
-	return fmt.Sprintf("mongodb://%s/?replicaSet=%s", strings.Join(m.Hosts(clusterDomain), ","), m.Name)
+	optionsString := m.GetOptionsString()
+
+	return fmt.Sprintf("mongodb://%s/?replicaSet=%s%s", strings.Join(m.Hosts(clusterDomain), ","), m.Name, optionsString)
```

Comment:
Since the `replicaSet` option is hardcoded in the connection string, what happens if it's provided as an additional option? Does it need to be filtered out of the computed `optionsString` ?

Reply:
Indeed, it will be duplicated in the connection string. The connection string would still work as any duplicate would be simply ignored. However, it makes sense that it should be filtered out.

Reply:
Would the connection string use the first option? We should make sure that is the case since we require using the `replicaSet` created with the operator through the automation config.

Reply:
Yes, it uses the first option.